src/bin/stats.rs

```diff
@@ -19,25 +22,35 @@ pub struct FrameSummary {
   pub size: usize,
   pub input_frameno: u64,
   pub frame_type: FrameType,
-  /// PSNR for Y, U, and V planes
-  pub psnr: Option<(f64, f64, f64)>,
+  /// Contains metrics such as PSNR, SSIM, etc.
+  pub metrics: Option<QualityMetrics>,
```

Comment:
I see now how you implemented `calculate_frame_metrics`, with each field being an optional--which makes sense, since there's the option to calculate only PSNR.

Given that, we can remove the `Option` from the field on this level. Sorry about the confusion. 😅